### PR TITLE
Bump LLVM version to 3.9

### PR DESCRIPTION
**Pull request by vchuravy**
This fixes the shlib build on windows and bumps LLVM to 3.9. I tested the shlib build on `3.8.0`, because we will need to adopt the following two patches to `3.9`
- `llvm-3.8.0_threads`
- `llvm-win64-reloc-dwarf`

### CODE REVIEW

**Comment by vchuravy** [comment]
@staticfloat We will need to update the homebrew llvm-version

**Comment by ararslan** [comment]
I suppose once this is merged we can go through and delete the portions of C/C++ code with `JL_LLVM_VERSION < 30900`?

**Comment by vchuravy** [comment]
Eventually maybe, but for now we will keep backwards compatibility to he!p debugging bugs and performance regressions.

**Comment by vchuravy** [comment]
I should add that I am not planning on adapting the patches for now (lack of time and lack of access to Windows). So I would welcome any help with that.

**Comment by ViralBShah** [comment]
Isn't it preferable to jump to 3.9.1 directly, even if it takes some more time, so that people don't spend time filing any issues with 3.8 that come up?

**Comment by tkelman** [comment]
We should try to skip straight to 3.9 if we can. Any change in LLVM version needs preparation work on CI and testing - if we can make that pass on 3.9 let's use it, only if we hit issues we could then try and see if 3.8 is any better.

**Comment by vchuravy** [comment]
Closing for now #19054 contains the important part of this PR and is submitted as a patch to upstream.

File: Makefile [hunk 1]
```diff
@@ -322,6 +322,10 @@ else
 $(eval $(call std_dll,gcc_s_seh-1))
 endif
 $(eval $(call std_dll,ssp-0))
+julia-src-release julia-debug-release julia_flisp.boot.inc.phony: | $(build_bindir)/LLVM.dll $(build_depsbindir)/LLVM.dll
+$(build_bindir)/LLVM.dll: | $(build_bindir) julia-deps
+$(build_depsbindir)/LLVM.dll: $(build_bindir)/LLVM.dll | $(build_depsbindir)
```
Comment (tkelman) [inline Makefile]: were you testing on msys2, or cygwin-cross? I haven't needed this yet in cygwin-cross, likely because of using llvm-config-host

Comment (vchuravy) [inline Makefile]: That was on cygwin, but it was an artifact of 6deb735.  #19054 solved it properly so we don't need this anymore.

**Pull request closed without merging**